netket/models/equivariant.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union, Optional, Tuple, Any, Callable, Iterable
+
+import numpy as np
+
+import jax
+from jax import numpy as jnp
+from flax import linen as nn
+
+from netket.hilbert import AbstractHilbert
+from netket.graph import AbstractGraph, SymmGroup
+from netket.utils.types import PRNGKeyT, Shape, DType, Array, NNInitFunc
+
+
+from netket import nn as nknn
+from netket.nn.initializers import lecun_complex, zeros, variance_scaling
+
+
+class GCNN(nn.Module):
+    """Implements a group convolutional neural network with symmetry
+    averaging in the last layer as described in Roth et al. 2021."""
+
+    symmetries: Callable[[], Array]
+    """permutations specifying symmetry group"""
+    group_algebra: Tuple
+    """Matrix specifying algebra of symmetry group given by SymmGroup"""
+    layers: int
+    """Number of layers (not including sum layer over output)"""
+    features: Union[Tuple, int]
+    """Number of features in each layer starting from the input. If
+    a single number is given, all layers will have the same number
+    of features"""
+    dtype: Any = np.float64
+    """The dtype of the weights."""
+    activation: Any = nknn.relu
+    """The nonlinear activation function between hidden layers."""
+    output_activation: Any = nknn.logcosh
+    """The nonlinear activation before the output."""
+    use_bias: bool = True
+    """if True uses a bias in all layers."""
+    precision: Any = None
+    """numerical precision of the computation see `jax.lax.Precision`for details."""
+    kernel_init: NNInitFunc = variance_scaling(1.0, "fan_in", "normal")
+    """Initializer for the Dense layer matrix."""
+    bias_init: NNInitFunc = zeros
+    """Initializer for the hidden bias."""
+
+    def setup(self):
+
+        self.n_symm, _ = self.symmetries().shape
```

Comment:
See the recent PR by @femtobit and changes to DenseSymm. Here you can just call 'symm.shape'

Reply:
Should be `np.asarray(self.symmetries).shape` as of the current master.